test/test_ops.py

```diff
@@ -345,6 +345,115 @@ def test_out(self, device, dtype, op):
 
+    @ops([op for op in op_db if op.aliases])
+    def test_aliases_consistency_eager(self, device, dtype, op):
+        # Currently, there is a code duplication based on
+        # test_variant_consistency_eager.
+        # To refactor those methods later.
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        for sample in samples:
+            expected_forward = op(*sample.input, *sample.args, **sample.kwargs)
+            # Computes expected backward
+            # NOTE: backward may fail for some dtypes
+            exception_during_backwards = False
+            expected_grad = None
+            try:
+                expected_forward.sum().backward()
+                expected_grad = sample.input.grad
+                sample.input.grad = None
+            except Exception as e:
+                exception_during_backwards = True
+
+            for a_op in op.aliases:
+
+                # Acquires variants to test
+                method = a_op.method_variant
+                inplace = a_op.inplace_variant
+                variants = (v for v in (a_op, method, inplace) if v is not None)
+
+                # Test eager consistency
+                for variant in variants:
+
+                    # Verifies that inplace operations that promote int->float fail
+                    #   on tensors with integer dtypes.
+                    if (variant is inplace and not torch.can_cast(expected_forward.dtype, dtype)):
+                        try:
+                            variant_forward = variant(*(clone_input_helper(input) for input in sample.input),
+                                                      *sample.args,
+                                                      **sample.kwargs)
+                        except Exception as e:
+                            continue
+                        self.fail("Inplace operation on integer tensor that should be promoted to float didn't fail!")
+                    # Compares variant's forward
+                    # Note: copy the tensor-type inputs when testing inplace operation
+                    variant_forward = variant(*(clone_input_helper(input) if variant is inplace else input
+                                                for input in sample.input),
+                                              *sample.args,
+                                              **sample.kwargs)
+                    self.assertEqual(variant_forward, expected_forward)
+
+                    # Compares variant's backward
+                    if variant is not inplace or op.test_inplace_grad:
+                        self.check_variant_backward(sample.input, variant_forward,
+                                                    expected_grad, exception_during_backwards)
+
+    @ops([op for op in op_db if op.aliases])
+    def test_aliases_consistency_jit(self, device, dtype, op):
```

Comment:
This test is good but I don't think it's testing that operator names are converted by the JIT:

https://github.com/pytorch/pytorch/blob/627a33125735a0683083f3e2b56e17fee80741b8/test/test_op_aliases.py#L228

That test is probably sufficient to ensure consistency, since the JIT is actually running the original function once the name is remapped.

What do you think about removing the jit consistency check and replacing it with the jit remapping check, @vfdev-5?

Reply:
Sounds great ! Thanks for pointing out !